cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/bitwarden"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	logLevel       string
+	configPath     string
+	bwUser         string
+	dryRun         bool
+	bwPasswordPath string
+	maxConcurrency int
+
+	config     []bitWardenItem
+	bwPassword string
+}
+
+type bitWardenItem struct {
+	ItemName   string         `json:"item_name"`
+	Field      fieldGenerator `json:"field,omitempty"`
+	Attachment fieldGenerator `json:"attachment,omitempty"`
+	Attribute  fieldGenerator `json:"attribute,omitempty"`
+}
+
+type fieldGenerator struct {
+	Name string `json:"name,omitempty"`
+	Cmd  string `json:"cmd,omitempty"`
+}
+
+const (
+	attributeTypePassword string = "password"
+)
+
+func parseOptions() options {
+	var o options
+	flag.CommandLine.BoolVar(&o.dryRun, "dry-run", false, "Whether to actually create the secrets with oc command")
+	flag.CommandLine.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	flag.CommandLine.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	flag.CommandLine.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	flag.CommandLine.StringVar(&o.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	flag.CommandLine.IntVar(&o.maxConcurrency, "concurrency", 1, "Maximum number of concurrent in-flight goroutines to BitWarden.")
+	if err := flag.CommandLine.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: %q", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	level, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid log level specified: %w", err)
+	}
+	logrus.SetLevel(level)
+	if o.bwUser == "" {
+		return fmt.Errorf("--bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("--bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("--config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions(secrets sets.String) error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+	secrets.Insert(o.bwPassword)
+
+	bytes, err = ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(bytes, &o.config)
+	if err != nil {
+		return err
+	}
+	return o.validateCompletedOptions()
+}
+
+func (o *options) validateCompletedOptions() error {
+	if o.bwPassword == "" {
+		return fmt.Errorf("--bw-password-file was empty")
+	}
+
+	for i, bwItem := range o.config {
+		if bwItem.ItemName == "" {
+			return fmt.Errorf("config[%d].itemName: empty key is not allowed", i)
+		}
+		switch bwItem.Attribute.Name {
```

Comment:
thats a lot easier expressed as `if bwItem.Attribute.Name != attributeTypePassword { error out }`

Reply:
its attributeTypePassword and the "" that I am checking for there. I can change it to 
`if bwItem.Attribute.Name != attributeTypePassword and bwItem.Attribute.Name!= ""` if that's more stylistically correct